crates/matrix-sdk-ui/tests/integration/timeline/queue.rs

```diff
@@ -277,9 +278,13 @@ async fn test_reloaded_failed_local_echoes_are_marked_as_failed() {
     // The error is not recoverable.
     assert!(!is_recoverable);
     // And it's properly pattern-matched.
-    assert_matches!(
-        error.as_client_api_error().unwrap().error_kind(),
-        Some(ruma::api::client::error::ErrorKind::TooLarge)
+    let msg = assert_matches!(
+        error,
+        QueueWedgeError::GenericApiError { msg } => msg
+    );
+    assert_eq!(
+        msg,
+        "the server returned an error: [413 / M_TOO_LARGE] Sounds like you have a lot to say!"
```

Comment:
(with respect to the previous comment, this would mean undoing this, since we could still pattern-match the right client API error here)

Reply:
Not for this one see https://github.com/matrix-org/matrix-rust-sdk/pull/4137#discussion_r1808911747
EventSendState is using `QueueWedgeError` and not the full `crate::Error`. Notice that for this particular error we propbably will want to add a proper variant for `QueueWedgeError` to have a proper media local echo feedback